### DISCUSSION TITLE
BH with Focusrite and Midi Keyboard to MainStage to Zoom

### DISCUSSION THREAD

**Discussion by Mehdibuntu** [ExistentialAudio/BlackHole#126] [General]
Hello, 
Thank you for your BlackHole plugin.
I installed it and I can't find how to make it work with Scarlet Focusrite (with microphone) and a midi keyboard. It's for live use on Zoom. I also have MainStage but I don't know if it's useful. 
I have always the voice with the mic but never the Keyboard. 
Thanks for your help and advice.

**Comment by devinroth** [comment]
How do you have everything setup?

**Comment by Mehdibuntu** [comment]
![image](https://user-images.githubusercontent.com/64665363/80846436-7219a180-8c0c-11ea-8143-07d7abe1f854.png)![image](https://user-images.githubusercontent.com/64665363/80846442-7940af80-8c0c-11ea-8c59-b0bd6f472eb2.png)![image](https://user-images.githubusercontent.com/64665363/80846445-7d6ccd00-8c0c-11ea-9cb1-fc3a755afac9.png)![image](https://user-images.githubusercontent.com/64665363/80846449-8362ae00-8c0c-11ea-9288-7236e3f104e3.png)![image](https://user-images.githubusercontent.com/64665363/80846450-86f63500-8c0c-11ea-9670-098b41651d3f.png)![image](https://user-images.githubusercontent.com/64665363/80846452-8bbae900-8c0c-11ea-9591-8f5477880165.png)

It work but not completely. Here in GB I can record each track.
And without reason all are disconnected. And now I can’t change the choice in GB.
(Here I use GB for check with simple app)

**Comment by Mehdibuntu** [comment]
I may have found something that works. 

 

I managed to get the midi keyboard and the microphone into Zoom. 

 

There's no latency on the musician side and in the headphone returns. 
 
The effects and reverb work.



I started a Zoom meeting

On the spectator side I can hear every instrument (voice and piano) but there are like sound cuts and also unpleasant interferences that I don't usually have in Zoom.
![image](https://user-images.githubusercontent.com/64665363/80849908-8fa13800-8c19-11ea-9dc8-5232e4f842c4.png)![image](https://user-images.githubusercontent.com/64665363/80849918-9b8cfa00-8c19-11ea-9e5b-1ba4438a3a31.png)![image](https://user-images.githubusercontent.com/64665363/80849921-9f208100-8c19-11ea-8f6b-8f42ae20cbe6.png)![image](https://user-images.githubusercontent.com/64665363/80849927-a6478f00-8c19-11ea-8702-e75fe4470d0c.png)![image](https://user-images.githubusercontent.com/64665363/80849932-ad6e9d00-8c19-11ea-971d-435b9a118f41.png)![image](https://user-images.githubusercontent.com/64665363/80849940-b52e4180-8c19-11ea-812a-aa75a8b90e1d.png)![image](https://user-images.githubusercontent.com/64665363/80849943-ba8b8c00-8c19-11ea-9281-bc26261cbd94.png)

**Comment by Mehdibuntu** [comment]
Someone can help me ?

**Comment by devinroth** [comment]
I'll take a closer look once I get a chance.

**Comment by devinroth** [comment]
Are you able to make the multi-output work with the default output? Set your speaker to the multi-output and route that to zoom like you are doing. If we can get that setup then we can work on adding the mic in.
